core/src/main/scala/org/apache/spark/HttpServer.scala

```diff
@@ -155,6 +156,7 @@ private[spark] class HttpServer(
       throw new ServerStateException("Server is already stopped")
     } else {
       server.stop()
+      Option(server.getThreadPool).collect { case x: LifeCycle => x }.foreach(_.stop())
```

Comment:
Interesting opportunity to talk about style -- yes I would have written it like @rxin suggests. It reads naturally: "if the pool is not null and is a LifeCycle, then call stop on it". The `Option` way splits the difference, and is OK by me, but is a little more obscure "an optional pool, collect them to themselves as LifeCycles, and then on all of them call stop". I also tend to think Scala syntax is overused.